CHANGELOG.md

```diff
@@ -1,3 +1,22 @@
+## v1.14.0 (15/11/2021)
+
+**Common**
+* **Work In Progress** Added LMDB library as a dependency. Currently not used, turned off.
+* Added handling of the `SSLException` error on Android as `IO_ERROR`.
+* Added handling of the `NSURLErrorCannotConnectToHost` error on iOS as `IO_ERROR`.
+* Added a new class `olp::cache::KeyGenerator`. It is used by the `olp-cpp-sdk-dataservice-read` module to generate cache keys for storing the data.
+* Removed the retry code from the Android network implementation. The retry logic is implemented in the `olp::client::OlpClient` class.
+* Fix various doxygen warnings.
+
+**olp-cpp-sdk-authentication**
+* Added the retry settings to the `olp::authentication::AuthenticationSettings`. Now the user can configure the `olp::authentication::AuthenticationClient` to use the provided retry settings for the online requests.
+* Added a new operator to the `olp::authentication::TokenProvider` that accepts the `olp::client::CancelationContext` as argument and returns `olp::authentication::TokenResponse`. Now the authentication requests could be canceled.
+* Deprecated the `olp::authentication::TokenProvider::operator()()` function. It will be removed by 10.2022.
+* Added a new `token_provider` to the `olp::client::OlpClientSettings`. New token provider accepts `olp::client::CancelationContext` and returns the `olp::client::OauthTokenResponse` response. Now the `CallApi` calls could be canceled during authentication.
+* Deprecated the `olp::authentication::AuthenticationSettings::provider` member. It will be removed by 10.2022.
```

Comment:
Merge with the above deprecation message, like:
"Deprecated `olp::authentication::TokenProvider::operator()()` method together with the `olp::authentication::AuthenticationSettings::provider` member. Both will be removed by 10.2022."